### PR TITLE
Filters implementation

### PR DESCRIPTION
**Pull request by Buuz135**

### CODE REVIEW

File: src/main/java/com/hrznstudio/titanium/api/filter/IFilter.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+    String getName();
+
+    boolean acceptsAsFilter(ItemStack filter);
+
+    void setFilter(int slot, ItemStack stack);
+
+    void setFilter(int slot, FilterSlot<T> filterSlot);
+
+    FilterSlot<T>[] getFilter();
```
Comment (SkySom) [inline src/main/java/com/hrznstudio/titanium/api/filter/IFilter.java]: I'd make it more obvious in the method name that is the slot.

File: src/main/java/com/hrznstudio/titanium/client/gui/addon/ItemstackFilterGuiAddon.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+import java.awt.*;
+import java.util.Objects;
+
+public class ItemstackFilterGuiAddon extends BasicGuiAddon implements IClickable {
+
+    private final ItemstackFilter filter;
+
+    public ItemstackFilterGuiAddon(ItemstackFilter filter) {
+        super(filter.getFilter()[0].getX(), filter.getFilter()[0].getY());
```
Comment (SkySom) [inline src/main/java/com/hrznstudio/titanium/client/gui/addon/ItemstackFilterGuiAddon.java]: Why is this based on the first slot?

File: src/main/java/com/hrznstudio/titanium/filter/ItemstackFilter.java [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    }
+
+    @Override
+    public boolean acceptsAsFilter(ItemStack filter) {
+        return true;
+    }
+
+    @Override
+    public void setFilter(int slot, ItemStack stack) {
+        filter[slot].setFilter(stack);
```
Comment (SkySom) [inline src/main/java/com/hrznstudio/titanium/filter/ItemstackFilter.java:59]: You should probably do some error checking here.

File: src/main/java/com/hrznstudio/titanium/client/gui/addon/ItemstackFilterGuiAddon.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+                    screen.renderTooltip(screen.getTooltipFromItem(filterSlot.getFilter()), mouseX - guiX, mouseY - guiY);
+                    GlStateManager.color4f(1, 1, 1, 1);
+                }
+            }
+        }
+    }
+
+    @Override
+    public void handleClick(Screen screen, int guiX, int guiY, double mouseX, double mouseY, int button) {
+        if (screen instanceof ContainerScreen && ((ContainerScreen) screen).getContainer() instanceof INetworkable) {
```
Comment (SkySom) [inline src/main/java/com/hrznstudio/titanium/client/gui/addon/ItemstackFilterGuiAddon.java:82]: I see now where you were asking about the locator stuff.

**Review by SkySom — CHANGES_REQUESTED** [review]

File: src/main/java/com/hrznstudio/titanium/client/gui/addon/ItemstackFilterGuiAddon.java (hunk 2 above)
Comment (Buuz135) [reply src/main/java/com/hrznstudio/titanium/client/gui/addon/ItemstackFilterGuiAddon.java]: Because the first slot is the 0, 0 coords of the filter

**Review by adrwa — APPROVED** [review]

**Pull request merged**